src/resolver.ts

```diff
@@ -3130,12 +3138,62 @@ export class Resolver extends DiagnosticEmitter {
               }
             }
             if (!fieldType) break; // did report above
+            if (existingField !== null) {
+              // visibility checks
+              let baseClass = <Class>base;
+              let thisFieldIsPubOrProt = !fieldPrototype.is(CommonFlags.PRIVATE);
+              let existingFieldIsPublic = existingField.isPublic;
+
+              if (thisFieldIsPubOrProt && !existingFieldIsPublic) {
```

Comment:
This seems to also catch `protected < protected`, which I guess would be OK?

Reply:
Argh, I forgot to add a test for that! 😅